pkg/controller/selector/selector.go

```diff
@@ -67,18 +67,13 @@ func (s *selector) SelectSupportedPoolForVolume(vol *model.VolumeSpec) (*model.S
 	// Generate filter request according to the rules defined in profile.
 	fltRequest := func(prf *model.ProfileSpec, vol *model.VolumeSpec) map[string]interface{} {
 		var filterRequest map[string]interface{}
+		filterRequest = prf.CustomProperties.GetCapabilitiesProperties()
 
-		if !prf.CustomProperties.IsEmpty() {
-			filterRequest = prf.CustomProperties
-
-			if v, ok := filterRequest["multiAttach"]; ok && v == "<is> true" {
-				log.Info("change volume multiAttach flag to be true.")
-				vol.MultiAttach = true
-			}
-
-		} else {
-			filterRequest = make(map[string]interface{})
+		if v, ok := filterRequest["multiAttach"]; ok && v == "<is> true" {
+			log.Info("change volume multiAttach flag to be true.")
+			vol.MultiAttach = true
```

Comment:
I don't know why vol.MultiAttach is set true here.

Reply:
Set to true here for volume status judgment，please see pkg/api/util/db.go line 249.
```
if vol.Status == model.VolumeAvailable {
		db.UpdateVolumeStatus(ctx, db.C, vol.Id, model.VolumeAttaching)
	} else if vol.Status == model.VolumeInUse {
		if vol.MultiAttach {
			db.UpdateVolumeStatus(ctx, db.C, vol.Id, model.VolumeAttaching)
		} else {
			msg := "volume is already attached or volume multiattach must be true if attach more than once"
			log.Error(msg)
			return nil, errors.New(msg)
		}
	} else {
		errMsg := "only the status of volume is available, attachment can be created"			errMsg := "only the status of volume is available, attachment can be created"
		log.Error(errMsg)		
		return nil, errors.New(errMsg)			
	}}
```

Reply:
I'm afraid the vol defined in selector.go is not stored in db.